tools/objc/libtool_check_unique.cc

```diff
@@ -32,43 +37,71 @@ string getBasename(const string &path) {
   return path.substr(idx + 1);
 }
 
-// Returns 0 if there are no duplicate basenames in the object files (both via
-// -filelist as well as shell args), 1 otherwise
-int main(int argc, const char *argv[]) {
+std::vector<string> readFile(const string path) {
+  std::vector<string> lines;
+  ifstream file(path);
+  string line;
+  while (std::getline(file, line)) {
+    if (!line.empty()) {
+      lines.push_back(line);
+    }
+  }
+
+  return lines;
+}
+
+unordered_set<string> parseArgs(std::vector<string> args) {
   unordered_set<string> basenames;
-  const regex libRegex = regex(".*\\.a$");
-  const regex noArgFlags =
-      regex("-static|-s|-a|-c|-L|-T|-D|-no_warning_for_no_symbols");
-  const regex singleArgFlags = regex("-arch_only|-syslibroot|-o");
-  // Set i to 1 to skip executable path
-  for (int i = 1; argv[i] != nullptr; i++) {
-    const string arg = argv[i];
+  for (auto it = args.begin(); it != args.end(); ++it) {
```

Comment:
nit: use `<` here in case a badly set of args would overrun?

Reply:
sounds like this isn't really best practice https://stackoverflow.com/a/6673775/902968

Reply:
i guess you have technically have UB either way, but the UB of over-incrementing an iterator seems trivial. anyways i'm fine either way

Reply:
that is, UB in the case of a buffer overrun